Example:Soccer Game 

You are off to soccer and want to be the goalkeeper but that depends on who will be coaching today:

with coach Sam, the probability of being the goalkeeper is 0.5
with coach Alex, the probability of being the goalkeeper is 0.3

Sam coaches more often, 6/10 games or 0.6 
prob that you will be goal keeper?

P(G|S) = 0.5 - Probability of being goalie, given sam
P(G|A) = 0.3 - probability of being goalie, given Alex
P(S) = 0.6 -  probability of Alex coaching

In [1]:
#import numpy 
import numpy as np

In [6]:
#How many times we run simulation 
n_runs = 100000

#initialize variables
SAM = 0
SAM_YES = 0
SAM_NO = 0 
ALEX =0
ALEX_YES = 0
ALEX_NO = 0
YES = 0
NO = 0

In [8]:
#loop that will simulate the process

for x in range(n_runs):
    # Sam (probability of sam being the coach)
    if np.random.random() < 0.6:
        SAM +=1
        # YES to Goal keep
        if np.random.random() < 0.5:
            SAM_YES += 1 
            YES += 1 
        #NO to goal keep
        else: 
            SAM_NO += 1 
            NO += 1 
    # Alex (prob of alex being coach)
    else: 
        ALEX += 1
        #Yes to goalie
        if np.random.random() < 0.3:
            ALEX_YES += 1 
            YES += 1 
        else:
            ALEX_NO += 1
            NO += 1 

In this simulation you are the goalkeeper and sam is the coach, for every time, Alex or Sam says yes, there is a counter for each, as well as Yes's in general. 
Create probabilities for all conditions: 

In [9]:
P_SAM = SAM / n_runs * 100
P_SAM_YES = SAM_YES / n_runs * 100
P_SAM_NO = SAM_NO / n_runs * 100
P_ALEX = ALEX / n_runs * 100
P_ALEX_YES = ALEX_YES / n_runs * 100
P_ALEX_NO = ALEX_NO / n_runs * 100
P_YES = YES / n_runs * 100
P_NO = NO / n_runs * 100

#print the probabilities
print(f'Sam is the coach: {P_SAM}%')
print(f'Sam is the coach and you are the goalkeeper: {P_SAM_YES}%')
print(f'Sam is the coach and you are not the goalkeeper: {P_SAM_NO}%')
print(f'Alex is the coach: {P_ALEX}%')
print(f'Alex is the coach and you are the goalkeeper: {P_ALEX_YES}%')
print(f'Alex is the coach and you are not the goalkeeper: {P_ALEX_NO}%')
print(f'You are the goalkeeper: {P_YES}%')
print(f'You are not the goalkeeper: {P_NO}')

Sam is the coach: 60.236000000000004%
Sam is the coach and you are the goalkeeper: 30.186%
Sam is the coach and you are not the goalkeeper: 30.049999999999997%
Alex is the coach: 39.763999999999996%
Alex is the coach and you are the goalkeeper: 11.848%
Alex is the coach and you are not the goalkeeper: 27.916%
You are the goalkeeper: 42.034%
You are not the goalkeeper: 57.965999999999994


In [10]:
#P(SAM | goalkeeper)
P_SAM_GOAL = P_SAM_YES / P_YES 
print(P_SAM_GOAL)

0.7181329400009516


Probability of you being the goalkeeper when sam is the coach, divided by the probability that you are the goalkeeper. 
    - Probability is about 72% 
    in about 72% of the games you're in as goal keeper, you coach is sam. 